srtcore/core.cpp

```diff
@@ -4427,7 +4460,25 @@ EConnectStatus srt::CUDT::processConnectResponse(const CPacket& response, CUDTEx
                      << "processConnectResponse: CONFUSED: expected UMSG_HANDSHAKE as connection not yet established, "
                         "got: "
                      << MessageTypeStr(response.getType(), response.getExtendedType()));
+
+            if (response.getType() == UMSG_SHUTDOWN)
+            {
+                LOGC(cnlog.Error,
+                        log << CONID() << "processConnectResponse: UMSG_SHUTDOWN received, rejecting connection.");
+                return CONN_REJECT;
+            }
```

Comment:
If SHUTDOWN is received, treat it as rejection. ✔️ 
Reason missing? Maybe ROGUE?

Reply:
Good point, althohgh I think the reject reason should be set already. It could be also done here as a fallback.